ming/schema.py

```diff
@@ -708,10 +710,50 @@ def _validate(self, value, **kw):
         except Exception as ex:
             raise Invalid(str(ex), value, None)
 
+
+class NumberDecimal(ParticularScalar):
+    """Validates value is compatible with :class:`bson.Decimal128`.
+
+    Useful for financial data that need arbitrary precision.
+
+    The argument `precision` specifies the number of decimal digits to evaluate.
+    Defaults to 6.
+
+    The argument `rounding` specifies the kind of rounding to be performed.
+    Valid values are `decimal.ROUND_DOWN`, `decimal.ROUND_HALF_UP`, `decimal.ROUND_HALF_EVEN`,
+    `decimal.ROUND_CEILING`, `decimal.ROUND_FLOOR`, `decimal.ROUND_UP`,
+    `decimal.ROUND_HALF_DOWN`, `decimal.ROUND_05UP`.
+    Defaults to `decimal.ROUND_HALF_DOWN`.
+
+    """
+    type = (int, float, Decimal, Decimal128)
+
+    def __init__(
+        self,
+        precision=6,
```

Comment:
Maybe we should support `precision=None` as it might be unexpected that it does quantization by default.
I guess in 90% of cases Decimals will be used for financial apps and thus rounding is expected.

Reply:
Fair enough, I'll make a fix for that :)

Reply:
How about the rounding though? I think it makes sense to have a default for rounding, but maybe we want to make that explicit too?

Reply:
It's fine, if you have `precision=None` it already disables the rounding as there is no rounding to do if you want to keep infinite precision.

Reply:
just added a commit that fixes the default behavior 😄